services/network/logicalnetwork/wssd.go

```diff
@@ -79,6 +79,52 @@ func (c *client) Delete(ctx context.Context, location, name string) error {
 	return err
 }
 
+func (c *client) Precheck(ctx context.Context, location string, logicalNetworks []*network.LogicalNetwork) (bool, error) {
+	request, err := getLogicalNetworkPrecheckRequest(location, logicalNetworks)
+	if err != nil {
+		return false, err
+	}
+	response, err := c.LogicalNetworkAgentClient.Precheck(ctx, request)
+	if err != nil {
+		return false, err
+	}
+	return getLogicalNetworkPrecheckResponse(response)
+}
+
+func getLogicalNetworkPrecheckRequest(location string, logicalNetworks []*network.LogicalNetwork) (*wssdcloudnetwork.LogicalNetworkPrecheckRequest, error) {
+	request := &wssdcloudnetwork.LogicalNetworkPrecheckRequest{}
+
+	protoLogicalNetworks := make([]*wssdcloudnetwork.LogicalNetwork, 0, len(logicalNetworks))
+
+	for _, logicalNetwork := range logicalNetworks {
+		// can logical network ever be nil here? what would be the meaning of that?
+		if logicalNetwork != nil {
+
+			// TODO (aweston): double check this
```

Comment:
This was one thing I wasn't sure about - all of the other getWssd* functions take location/group as separate parameters, but not this one. @mjethwa-msft can you give any context on the intent of that logic in getWssdLogicalNetwork/getLogicalNetworkRequest?

Reply:
Logical network is a location based entitity

Reply:
Sorry for the confusion, the part I was unsure about is that all the other resources have location or group as a separate param to the equivalent conversion function, whereas here it is not a param, but expected to be filled out in the SDK struct. I was not sure if there was any reason that was done, since the top level functions still have location as a separate parameter. The getWssdLogicalNetwork function will fail if it is nil, so it seems odd to have two different ways to pass it. Mostly just wanted to check if there was something I was missing here, and whether it is always expected to be filled out in the SDK struct itself.

To give an example:

func getWssdLogicalNetwork(c *network.LogicalNetwork) 
vs. virtual machine image where group is a separate parameter
func getWssdVirtualMachineImage(c *compute.VirtualMachineImage, **groupName string)**

Reply:
Ok, sorry got the question wrong. @mjethwa-msft Was there a reason for this? 

Today Lnet are created from Moc_operator only, so may be they always pass the location in the struct. I thing it was just an assumption made

Reply:
Going to go ahead and merge as is and will follow up separately. I think what I have implemented is safe and reasonable, just would be good to clarify the invariant here (if any) so we could simplify the logic.